Source/UnitTests/UnitTests.vcxproj

```diff
@@ -45,8 +53,9 @@
         The following libs are needed since we pull in pretty much the entire
         dolphin codebase.
         -->
-      <AdditionalLibraryDirectories>$(ExternalsDir)ffmpeg\lib;%(AdditionalLibraryDirectories)</AdditionalLibraryDirectories>
-      <AdditionalDependencies>avrt.lib;iphlpapi.lib;winmm.lib;setupapi.lib;opengl32.lib;glu32.lib;rpcrt4.lib;comctl32.lib;avcodec.lib;avformat.lib;avutil.lib;swresample.lib;swscale.lib;Shlwapi.lib;%(AdditionalDependencies)</AdditionalDependencies>
+      <AdditionalDependencies>avrt.lib;iphlpapi.lib;winmm.lib;setupapi.lib;rpcrt4.lib;comctl32.lib;Shlwapi.lib;%(AdditionalDependencies)</AdditionalDependencies>
+      <AdditionalDependencies Condition="'$(Platform)'=='x64'">opengl32.lib;avcodec.lib;avformat.lib;avutil.lib;swresample.lib;swscale.lib;%(AdditionalDependencies)</AdditionalDependencies>
```

Comment:
excluding the ffmpeg libraries seems to only be required because dolphin doesn't build this External from source. 
Here is a third party repo including x64 and arm64 builds of ffmpeg: https://github.com/Aftnet/FFmpegWin Maybe dolphin could use it or something like it? I was never a fan of committing binaries to externals... :/

Reply:
p.s. ffmpeg was historically OK'd to be committed as binaries as it required the WDK to build, which doesn't make sense to require all dolphin devs to deal with when they probably don't care at all about it. However, judging from the build script in that repo, the WDK may not be required anymore. It's probably worth revisiting if we can just build ffmpeg from source and/or make it optional on windows.

Reply:
@RisingFog is the contributor of ffmpeg builds for dolphin, maybe they have some input here.
They have some franken-fork of ffmpeg here: https://github.com/RisingFog/dolphin-ffmpeg (mainline has more recent updates: https://github.com/hrydgard/ppsspp-ffmpeg ), however since the current version in Externals (from https://github.com/dolphin-emu/dolphin/pull/4989 ) is higher than the one in RisingFog's repo, it's not clear how related they are.

@RisingFog it would be nice to know how whatever's currently in dolphin was generated.

Reply:
d'oh, missed that there's a dolphin branch there: https://github.com/RisingFog/dolphin-ffmpeg/commits/dolphin

there's also some compile log in the object files, like:
```
C:\Program Files (x86)\Microsoft Visual Studio 14.0\VC\BIN\amd64\cl.exe
-IC:\msys64\home\silen\dolphin-ffmpeg-dolphin -IC:\msys64\home\silen\dolphin-ffmpeg-dolphin\ -D_ISOC99_SOURCE -D_FILE_OFFSET_BITS=64 -D_LARGEFILE_SOURCE -D_WIN32_WINNT=0x0502 -DHAVE_AV_CONFIG_H -nologo -D_USE_MATH_DEFINES -D_CRT_SECURE_NO_WARNINGS
 -D_CRT_NONSTDC_NO_WARNINGS -MT -IC:\msys64\home\silen\dolphin-ffmpeg-dolphin\WindowsInclude -GS- -Z7 -W4 -wd4244 -wd4127 -wd4018 -wd4389 -wd4146 -wd4057 -wd4204 -wd4706 -wd4305 -wd4152 -wd4324 -we4013 -wd4100 -wd4214 -wd4307 -wd4273
 -wd4554 -wd4701 -O2 -SSAOptimizer- -c -I"C:\Program Files (x86)\Microsoft Visual Studio 14.0\VC\INCLUDE" -I"C:\Program Files (x86)\Microsoft Visual Studio 14.0\VC\ATLMFC\INCLUDE" -I"C:\Program Files (x86)\Windows Kits\10\include\10.0.10586.0\ucrt"
 -I"C:\Program Files (x86)\Windows Kits\NETFXSDK\4.6.1\include\um" -I"C:\Program Files (x86)\Windows Kits\10\include\10.0.10586.0\shared" -I"C:\Program Files (x86)\Windows Kits\10\include\10.0.10586.0\um" -I"C:\Program Files (x86)\Windows
 Kits\10\include\10.0.10586.0\winrt" -TC -X
```
however it still doesn't match what's in the repo (https://github.com/RisingFog/dolphin-ffmpeg/commit/6e0cf29eb02a1ef98c89d9432ca16dbf595011fa#diff-98ad90cfa64c27ce76009818fc95c8a5)

there is however this, which is maybe enough to repro:
```
--extra-cflags='-MT -IWindowsInclude -GS-' --toolchain=msvc --prefix=./Windows/x86_64 --arch=x86_64 --disable-avdevice --disable-programs --disable-avfilter --disable-postproc --disable-doc --disable-pthreads --enable-w32threads --disable-network --disable-everything --disable-encoders --disable-muxers --disable-hwaccels --disable-parsers --disable-protocols --enable-static --disable-shared --enable-muxer=avi --enable-encoder=ffv1 --enable-encoder=mpeg4 --enable-protocol=file
```

Reply:
was able to build ffmpeg under wsl with a bit of a hack:

```diff
diff --git a/configure b/configure                                                                                                                                                                                                             index 34c2adb4a4..e49af5ee65 100755                                                                                                                                                                                                            --- a/configure                                                                                                                                                                                                                                +++ b/configure                                                                                                                                                                                                                                @@ -4607,7 +4607,7 @@ probe_cc(){                                                                                                                                                                                                                   elif $_cc -nologo- 2>&1 | grep -q Microsoft || { $_cc -v 2>&1 | grep -q clang && $_cc -? > /dev/null 2>&1; }; then                                                                                                                                 _type=msvc                                                                                                                                                                                                                                     _ident=$($_cc 2>&1 | head -n1 | tr -d '\r')                                                                                                                                                                                           -        _DEPCMD='$(DEP$(1)) $(DEP$(1)FLAGS) $($(1)DEP_FLAGS) $< 2>&1 | awk '\''/including/ { sub(/^.*file: */, ""); gsub(/\\/, "/"); if (!match($$0, / /)) print "$@:", $$0 }'\'' > $(@:.o=.d)'                                               +        _DEPCMD='$(DEP$(1)) $(DEP$(1)FLAGS) $($(1)DEP_FLAGS) $< 2>&1 | awk '\''/including/ { sub(/^.*file: */, ""); gsub(/\\/, "/"); gsub(/C:/, "/mnt/c/"); if (!match($$0, / /)) print "$@:", $$0 }'\'' > $(@:.o=.d)'                                 _DEPFLAGS='$(CPPFLAGS) $(CFLAGS) -showIncludes -Zs'                                                                                                                                                                                            _cflags_speed="-O2"                                                                                                                                                                                                                            _cflags_size="-O1"
```
hopefully there's a more proper way

Reply:
I've built ffmpeg for x64 and arm64 and made a repo designed to function like the existing dolphin Qt submodule: https://github.com/shuffle2/ext-win-ffmpeg
The build source is here: https://github.com/shuffle2/FFmpeg/commits/dolphin
I would expect dolphin project to fork both these repos and effectively take them over. Hopefully it simplifies dealing with ffmpeg in the future.

I tested that dolphin x64 + arm64 builds against these build outputs.

Reply:
btw, the configure outputs:
```
install prefix            ./ext-win-ffmpeg/ARM64
source path               .
C compiler                cl.exe
C library                 msvcrt
host C compiler           gcc
host C library            glibc
ARCH                      aarch64 (generic)
big-endian                no
runtime cpu detection     yes
NEON enabled              yes
VFP enabled               yes
debug symbols             yes
strip symbols             no
optimize for size         no
optimizations             yes
static                    yes
shared                    no
postprocessing support    no
network support           no
threading support         w32threads
safe bitstream reader     yes
texi2html enabled         no
perl enabled              yes
pod2man enabled           yes
makeinfo enabled          no
makeinfo supports HTML    no

External libraries:
schannel

External libraries providing hardware acceleration:
d3d11va                 dxva2

Libraries:
avcodec                 avformat                avutil                  swresample              swscale

Programs:

Enabled decoders:

Enabled encoders:
ffv1                    h263                    mpeg4

Enabled hwaccels:

Enabled parsers:

Enabled demuxers:

Enabled muxers:
avi

Enabled protocols:
file

Enabled filters:

Enabled bsfs:
null

Enabled indevs:

Enabled outdevs:

License: LGPL version 2.1 or later
```
```
install prefix            ./ext-win-ffmpeg/x64
source path               .
C compiler                cl.exe
C library                 msvcrt
ARCH                      x86 (generic)
big-endian                no
runtime cpu detection     yes
standalone assembly       yes
x86 assembler             yasm
MMX enabled               yes
MMXEXT enabled            yes
3DNow! enabled            yes
3DNow! extended enabled   yes
SSE enabled               yes
SSSE3 enabled             yes
AESNI enabled             yes
AVX enabled               yes
AVX2 enabled              yes
AVX-512 enabled           yes
XOP enabled               yes
FMA3 enabled              yes
FMA4 enabled              yes
i686 features enabled     yes
CMOV is fast              yes
EBX available             no
EBP available             no
debug symbols             yes
strip symbols             no
optimize for size         no
optimizations             yes
static                    yes
shared                    no
postprocessing support    no
network support           no
threading support         w32threads
safe bitstream reader     yes
texi2html enabled         no
perl enabled              yes
pod2man enabled           yes
makeinfo enabled          no
makeinfo supports HTML    no

External libraries:
schannel

External libraries providing hardware acceleration:
d3d11va                 dxva2

Libraries:
avcodec                 avformat                avutil                  swresample              swscale

Programs:

Enabled decoders:

Enabled encoders:
ffv1                    h263                    mpeg4

Enabled hwaccels:

Enabled parsers:

Enabled demuxers:

Enabled muxers:
avi

Enabled protocols:
file

Enabled filters:

Enabled bsfs:
null

Enabled indevs:

Enabled outdevs:

License: LGPL version 2.1 or later
```

Reply:
@shuffle2 I was using msys to cross-compile the libraries, but this was some time ago.

Reply:
@RisingFog thanks :) was mainly looking for configure options, which was helpfully embedded as a string into the object files, so nothing more to do here.